data/memory.go

```diff
@@ -54,50 +105,56 @@ func (m *MemoryConnector) Get(ctx context.Context, index, partitionKey, rangeKey
 	}
 
 	key := fmt.Sprintf("%s:%s", partitionKey, rangeKey)
-	value, ok := m.cache.Get(key)
+	m.logger.Debug().Str("key", key).Msg("getting item from memory")
+
+	item, ok := m.cache.Get(key)
 	if !ok {
 		return "", common.NewErrRecordNotFound(fmt.Sprintf("PK: %s RK: %s", partitionKey, rangeKey), MemoryDriverName)
 	}
-	return value, nil
+
+	// Check if item has expired
+	if item.expiresAt != nil && time.Now().After(*item.expiresAt) {
+		m.cache.Remove(key)
+		return "", common.NewErrRecordNotFound(fmt.Sprintf("PK: %s RK: %s", partitionKey, rangeKey), MemoryDriverName)
+	}
+
+	return item.value, nil
 }
 
 func (m *MemoryConnector) getWithWildcard(_ context.Context, _, partitionKey, rangeKey string) (string, error) {
 	key := fmt.Sprintf("%s:%s", partitionKey, rangeKey)
 	for _, k := range m.cache.Keys() {
 		if common.WildcardMatch(key, k) {
-			value, _ := m.cache.Get(k)
-			return value, nil
+			item, _ := m.cache.Get(k)
+			// Check expiration
+			if item.expiresAt != nil && time.Now().After(*item.expiresAt) {
+				m.cache.Remove(k)
+				continue
+			}
+			return item.value, nil
 		}
 	}
 	return "", common.NewErrRecordNotFound(fmt.Sprintf("PK: %s RK: %s", partitionKey, rangeKey), MemoryDriverName)
 }
 
-func (m *MemoryConnector) Delete(ctx context.Context, index, partitionKey, rangeKey string) error {
-	if strings.HasSuffix(partitionKey, "*") || strings.HasSuffix(rangeKey, "*") {
-		return m.deleteWithWildcard(ctx, index, partitionKey, rangeKey)
-	}
-
-	key := fmt.Sprintf("%s:%s", partitionKey, rangeKey)
-	m.cache.Remove(key)
-	return nil
-}
-
-func (m *MemoryConnector) deleteWithWildcard(_ context.Context, _, partitionKey, rangeKey string) error {
-	prefixPK := strings.TrimSuffix(partitionKey, "*")
-	prefixRK := strings.TrimSuffix(rangeKey, "*")
+func (m *MemoryConnector) cleanupExpired() {
```

Comment:
shouldn't we return an `error` result here from this func?

Reply:
no error is really expected from here afaict 🤔

Reply:
understood. but I saw an error was expected for the same func in `PG` driver, that is why I was curious, but maybe there is different